src/test/java/qna/repository/AnswerRepositoryTest.java

```diff
@@ -0,0 +1,103 @@
+        assertAll(
+                () -> assertThat(answer.getId()).isEqualTo(expect.getId()),
+                () -> assertThat(answer.getWriterId()).isEqualTo(expect.getWriterId()),
+                () -> assertThat(answer.getQuestionId()).isEqualTo(expect.getQuestionId()),
+                () -> assertThat(answer.getContents()).isEqualTo(expect.getContents()),
+                () -> assertThat(answer.isDeleted()).isEqualTo(expect.isDeleted())
+        );
+    }
+
+    @Test
+    @DisplayName("answer 삭제(deleted 상태 변경)")
+    void answer_delete() {
+        Answer answer = new Answer(UserTest.JAVAJIGI, QuestionTest.Q1, "contents");
+        answers.saveAndFlush(answer);
+        answer.setDeleted(true);
+        entityManager.flush();
+        entityManager.clear();
+        Answer expect = answers.findById(answer.getId()).orElseThrow(NotFoundException::new);
+        assertTrue(expect.isDeleted());
+    }
+
+    @Test
+    @DisplayName("question id에 일치하는 삭제되지 않은 Answer을 조회")
+    void find_question_id_and_delete_false() {
+        Question question = questions.saveAndFlush(new Question("title", "contents"));
+        Answer answer = new Answer(UserTest.JAVAJIGI, question, "contents");
+        answers.saveAndFlush(answer);
+        entityManager.clear();
+        List<Answer> expect = answers.findByQuestionIdAndDeletedFalse(question.getId());
+        assertThat(expect).hasSize(1);
+        expect.forEach(v -> assertEquals(answer, v));
+    }
+
+    @Test
+    @DisplayName("question id에 일치하는 삭제되지 않은 Answer을 조회(해당 결과 없음 일차하는 questionId가 없음)")
+    void find_question_id_and_delete_false_return_nothing() {
+        Question question = questions.saveAndFlush(new Question("title", "contents"));
+        Answer answer = new Answer(UserTest.JAVAJIGI, question, "contents");
+        answers.saveAndFlush(answer);
+        entityManager.clear();
+        long diffQuestionId = 3L;
+        List<Answer> expect = answers.findByQuestionIdAndDeletedFalse(diffQuestionId);
+        assertThat(expect).isEmpty();
+    }
+
+    @Test
+    @DisplayName("answer id에 일치하는 삭제되지 않은 Answer을 조회")
+    void find_answer_id_and_delete_false() {
+        Answer answer = new Answer(UserTest.JAVAJIGI, QuestionTest.Q1, "contents");
+        answers.saveAndFlush(answer);
+        entityManager.clear();
+        Answer expect = answers.findByIdAndDeletedFalse(answer.getId()).orElseThrow(NotFoundException::new);
+        assertThat(answer).isEqualTo(expect);
+    }
+
+    @Test
+    @DisplayName("answer id에 일치하는 삭제되지 않은 Answer을 조회(삭제된 Answer 이라 조회결과 없음)")
+    void find_answer_id_and_delete_false_return_nothing() {
+        Answer answer = new Answer(UserTest.JAVAJIGI, QuestionTest.Q1, "contents");
+        answer.setDeleted(true);
+        answers.saveAndFlush(answer);
+        entityManager.clear();
+        assertThatThrownBy(() -> answers.findByIdAndDeletedFalse(answer.getId()).orElseThrow(NotFoundException::new))
+                .isInstanceOf(NotFoundException.class);
```

Comment:
다른 부분에서 `Optional` 검증을 잘 사용해주셔서 잘 아시겠지만 
조금 어색해보여서 코멘트 남겨드릴게요!! 😅
```suggestion
        assertThat(answers.findByIdAndDeletedFalse(answer.getId())).isEmpty();
```

Reply:
넵넵 이부분 다음 step에 수정해서 반영하겠습니다. 😄